src/randoop/operation/FieldSetter.java

```
@@ -119,20 +116,23 @@ public void appendCode(Variable newVar, List<Variable> inputVars, StringBuilder
     int index = inputVars.size() - 1;
 
     //TODO this is duplicate code from RMethod - should factor out behavior
-    Operation statementCreatingVar = inputVars.get(index).getDeclaringStatement();
-    if (!GenInputsAbstract.long_format && ExecutableSequence.canUseShortFormat(statementCreatingVar )) {
-      Object val = ((NonreceiverTerm) statementCreatingVar).getValue();
-      b.append(PrimitiveTypes.toCodeString(val));
+    Statement statementCreatingVar = inputVars.get(index).getDeclaringStatement();
+    if (!GenInputsAbstract.long_format) {
+      String shortForm = statementCreatingVar.getShortForm();
+      if (shortForm != null) {
```

Same comment as before about silently dropping a statement with a null shortForm.